daemon/src/maker_cfd.rs

```diff
@@ -603,17 +607,18 @@ where
             .address()
             .expect("actor to be able to give address to itself");
 
-        tokio::spawn(async move {
+        let task = async move {
             let dlc = contract_future.await;
 
-            this.do_send_async(CfdSetupCompleted { order_id, dlc })
-                .await
-        });
+            log_error!(this.do_send_async(CfdSetupCompleted { order_id, dlc }))
+        }
+        .spawn_with_handle();
```

Comment:
I think this is unnecessary, this task will end by itself sooner or later. I think the idea of using `spawn_with_handle` should only be applied to tasks that contain endless loops.

Reply:
I agree that main use-case is dealing with tasks containing endless loops, however in this case cleanup during shutdown could deal with it in a nice way IMHO (especially that we already have a state for in-flight setup, we're making the code clearer this way for a reader)